readpass.c

```diff
@@ -183,10 +183,19 @@ read_passphrase(const char *prompt, int flags)
 			askpass = getenv(SSH_ASKPASS_ENV);
 		else
 			askpass = _PATH_SSH_ASKPASS_DEFAULT;
+
+#ifdef WINDOWS
```

Comment:
DISPLAY environment isn't applicable for windows?
Change the logic to have line 188 to197 for windows.
Disable and don't touch the existing code for windows.

This makes the users not to worry about additional DISPLAY environment variable which is not applicable for windows.

Reply:
`DISPLAY` use is required on the source side for X11 forwarding to work. It points to an X Display Server of the user's choosing.

Basically,  `ssh` connects to the configured X Display Server and tunnels traffic to/from the target. `sshd` on the target side will automatically set up `DISPLAY` for application use.

So the commit only introduces one change: On Windows, it stops forcing the user to use ASKPASS if `SSH_ASKPASS` is not defined. If it is, business as usual. No impact to non-Windows platforms.

Am I missing something?

Reply:
I am not familiar with the x11 forwarding. Could you please share the steps on both the client and server so I can test?

Reply:
Absolutely, happy to share. It's way easier than it sounds too.

Try this:
1. On Windows, install [XMing](https://sourceforge.net/projects/xming/)
   * Chose to display programs in `Multiple windows` and keep the Display Number at `-1`.
   * Keep all other defaults
2. Start the display server, if not already running (`XLaunch` in Start)
3. In a PowerShell window, issue the following commands (replacing text as needed):
    ```
    $env:DISPLAY="localhost:0.0"
    ssh -Y user@host
    ```
4. On the target (assumed Linux with working package manager), install `xorg-x11-apps` package.
5. On the target, start `xclock` and observe window appears on Windows desktop!

Reply:
Thanks. I worked.
I was puzzled how the request is forwarded to xming server when we use ssh -Y user@host.
ssh client logs shows there is no xauth program but still they are able to forward the X  data to xming.

debug1: No xauth program.
Warning: No xauth data; using fake authentication data for X11 forwarding.
debug1: Requesting X11 forwarding with authentication spoofing.
debug2: channel 0: request x11-req confirm 1

Reply:
Great, so now to reproduce the issue this PR fixes:
1. Ensure `C:\dev\tty` doesn't exist
2. Ensure `DISPLAY` is set, then `ssh -Y user@host`
3. Observe authentication failure

The failure is a result of ssh passing the check at line 181 (use_askpass is 1 because we don't have a tty), then entering a block of code that assumes ASKPASS is present, which fails.

Reply:
I got the need for the fix.

I didn't understand how the data is forwarded as we didn't specify the xauth program when we do ssh -Y user@host.
debug1: No xauth program.
Warning: No xauth data; using fake authentication data for X11 forwarding.
debug1: Requesting X11 forwarding with authentication spoofing.
debug2: channel 0: request x11-req confirm 1

Also one observation, GUI doesn't display colors on windows side.

Reply:
Should see colors, try `xclock -bg pink`.